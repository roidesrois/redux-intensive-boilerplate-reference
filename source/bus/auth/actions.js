// Types
import { types } from './types';

export const authActions = {
    // Sync
    initialize: () => ({
        type: types.INITIALIZE,
    }),
    authenticate: () => ({
        type: types.AUTHENTICATE,
    }),
    logout: () => ({
        type: types.LOGOUT,
    }),

    // Async
    initializeAsync: () => ({
        type: types.INITIALIZE_ASYNC,
    }),
    loginAsync: (credentials) => ({
        type:    types.LOGIN_ASYNC,
        payload: credentials,
    }),
    authenticateAsync: () => ({
        type: types.AUTHENTICATE_ASYNC,
    }),
    signupAsync: (userData) => ({
        type:    types.SIGNUP_ASYNC,
        payload: userData,
    }),
    logoutAsync: () => ({
        type: types.LOGOUT_ASYNC,
    }),
};
