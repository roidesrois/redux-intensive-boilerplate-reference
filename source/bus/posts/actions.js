// Types
import { types } from './types';

export const postsActions = {
    fillPosts: (posts) => ({
        type:    types.FILL_POSTS,
        payload: posts,
    }),
    createPost: (post) => ({
        type:    types.CREATE_POST,
        payload: post,
    }),
    removePost: (postId) => ({
        type:    types.REMOVE_POST,
        payload: postId,
    }),
    likePost: (likedPostData) => ({
        type:    types.LIKE_POST,
        payload: likedPostData,
    }),
    unlikePost: (unlikedPostData) => ({
        type:    types.UNLIKE_POST,
        payload: unlikedPostData,
    }),
    clearPosts: () => ({
        type: types.CLEAR_POSTS,
    }),

    // Async
    fetchPostsAsync: () => ({
        type: types.FETCH_POSTS_ASYNC,
    }),
    createPostAsync: (comment) => ({
        type:    types.CREATE_POST_ASYNC,
        payload: comment,
    }),
    removePostAsync: (postId) => ({
        type:    types.REMOVE_POST_ASYNC,
        payload: postId,
    }),
    likePostAsync: (postId) => ({
        type:    types.LIKE_POST_ASYNC,
        payload: postId,
    }),
    unlikePostAsync: (postId) => ({
        type:    types.UNLIKE_POST_ASYNC,
        payload: postId,
    }),
};
