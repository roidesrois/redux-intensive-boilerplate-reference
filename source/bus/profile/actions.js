// Types
import { types } from './types';

export const profileActions = {
    // Sync
    fillProfile: (profile) => ({
        type:    types.FILL_PROFILE,
        payload: profile,
    }),
    updateAvatar: (newAvatarUrl) => ({
        type:    types.UPDATE_AVATAR,
        payload: newAvatarUrl,
    }),
    clearProfile: () => ({
        type: types.CLEAR_PROFILE,
    }),

    // Async
    updateNameAsync: (newName) => ({
        type:    types.UPDATE_NAME_ASYNC,
        payload: newName,
    }),
    updateAvatarAsync: (newAvatar) => ({
        type:    types.UPDATE_AVATAR_ASYNC,
        payload: newAvatar,
    }),
    updatePasswordAsync: (newPassword) => ({
        type:    types.UPDATE_PASSWORD_ASYNC,
        payload: newPassword,
    }),
};
