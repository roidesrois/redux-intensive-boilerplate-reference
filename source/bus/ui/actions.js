// Core
import { v4 } from 'uuid';

// Instruments
import { types } from './types';

export const uiActions = {
    setFetchingState: (state) => ({
        type:    types.SET_FETCHING_STATE,
        payload: state,
    }),
    emitError: (error, meta = null) => ({
        type:    types.EMIT_ERROR,
        payload: error,
        error:   true,
        meta,
    }),
    showNotification: (message, type = 'info', source = '') => ({
        type:    types.SHOW_NOTIFICATION,
        payload: {
            id: v4(),
            message,
            type,
            source,
        },
    }),
    hideNotification: () => ({
        type: types.HIDE_NOTIFICATION,
    }),
    setOnlineState: (state) => ({
        type:    types.SET_ONLINE_STATE,
        payload: state,
    }),
};
