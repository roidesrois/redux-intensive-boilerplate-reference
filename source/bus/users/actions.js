// Types
import { types } from './types';

export const usersActions = {
    // Sync
    fillUsers: (users) => ({
        type:    types.FILL_USERS,
        payload: users,
    }),
    clearUsers: () => ({
        type: types.CLEAR_USERS,
    }),

    // Async
    fetchUsersAsync: () => ({
        type: types.FETCH_USERS_ASYNC,
    }),
};
