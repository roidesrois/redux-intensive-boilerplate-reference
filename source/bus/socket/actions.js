// Instruments
import { socket } from '../../init/socket';
import { uiActions } from '../ui/actions';
import { postsActions } from '../posts/actions';

export const socketActions = Object.freeze({
    listenConnection: () => (dispatch) => {
        socket.on('connect', () => {
            dispatch(uiActions.setOnlineState(true));
        });

        socket.on('disconnect', () => {
            dispatch(uiActions.setOnlineState(false));
        });
    },

    listenPosts: () => (dispatch, getState) => {
        socket.on('create', (event) => {
            const { data: post } = JSON.parse(event);

            dispatch(postsActions.createPost(post));
        });

        socket.on('remove', (event) => {
            const { data: postId } = JSON.parse(event);

            dispatch(postsActions.removePost(postId));
        });

        socket.on('like', (event) => {
            const { data, meta } = JSON.parse(event);

            if (meta.action === 'like') {
                const liker = getState()
                    .users.find((user) => user.get('id') === data.userId)
                    .delete('avatar');

                dispatch(
                    postsActions.likePost({
                        postId: data.postId,
                        liker,
                    })
                );
            } else {
                dispatch(postsActions.unlikePost(data));
            }
        });
    },
});
