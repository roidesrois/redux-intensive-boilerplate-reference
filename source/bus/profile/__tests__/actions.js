// Core
import { profileActions } from '../actions';

describe('profile actions:', () => {
    // Sync
    test('fillProfile snapshot', () => {
        expect(profileActions.fillProfile(__.userProfile)).toMatchSnapshot();
    });

    test('updateAvatar snapshot', () => {
        expect(profileActions.updateAvatar(__.url)).toMatchSnapshot();
    });

    test('clearProfile snapshot', () => {
        expect(profileActions.clearProfile()).toMatchSnapshot();
    });

    // Async
    test('updateNameAsync snapshot', () => {
        expect(
            profileActions.updateNameAsync(__.userProfile),
        ).toMatchSnapshot();
    });

    test('updateAvatar snapshot', () => {
        expect(
            profileActions.updateAvatarAsync(__.userProfile),
        ).toMatchSnapshot();
    });

    test('updatePasswordAsync snapshot', () => {
        expect(
            profileActions.updatePasswordAsync(__.userProfile),
        ).toMatchSnapshot();
    });
});
