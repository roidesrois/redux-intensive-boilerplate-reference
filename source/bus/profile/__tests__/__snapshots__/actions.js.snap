// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile actions: clearProfile snapshot 1`] = `
Object {
  "type": "CLEAR_PROFILE",
}
`;

exports[`profile actions: fillProfile snapshot 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "Walter",
    "id": "TEST_ID",
    "lastName": "White",
    "token": "TEST_TOKEN",
  },
  "type": "FILL_PROFILE",
}
`;

exports[`profile actions: updateAvatar snapshot 1`] = `
Object {
  "payload": "https://www.url.com",
  "type": "UPDATE_AVATAR",
}
`;

exports[`profile actions: updateAvatar snapshot 2`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "Walter",
    "id": "TEST_ID",
    "lastName": "White",
    "token": "TEST_TOKEN",
  },
  "type": "UPDATE_AVATAR_ASYNC",
}
`;

exports[`profile actions: updateNameAsync snapshot 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "Walter",
    "id": "TEST_ID",
    "lastName": "White",
    "token": "TEST_TOKEN",
  },
  "type": "UPDATE_NAME_ASYNC",
}
`;

exports[`profile actions: updatePasswordAsync snapshot 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "Walter",
    "id": "TEST_ID",
    "lastName": "White",
    "token": "TEST_TOKEN",
  },
  "type": "UPDATE_PASSWORD_ASYNC",
}
`;
